dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -190,6 +201,8 @@ def test_add_profilers(self, *mocks):
     @mock.patch('dataprofiler.profilers.profile_builder.'
                 'ColumnDataLabelerCompiler')
     @mock.patch('dataprofiler.profilers.profile_builder.DataLabeler')
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'StructuredProfiler._update_correlation')
```

Comment:
why do we need this here? shouldn't we enforce correlation in the stream to be `None` or what we think it should be?

Reply:
Or is this being done elsewhere?

Reply:
this test uses mock, then 'statistics' is not updated, therefore I had to use mock for update_correlation. The merge is tested in test_merge_correlation

Reply:
The concern is that merge != stream. We need to test the condition when we stream data the the correlation gets set to `None`

Reply:
Also, maybe instead of mocking out update, you mock out get? and set the correlation to a specific value?

Then you can test the stream functionality:
```
if self.total_samples > 0:
            # currently return None with the existing correlation
            # TODO: implement the update with the existing correlation
            warnings.warn("Currently, the updating correlations is disabled "
                          "and will remain unchanged from its current value. "
                          "Updating correlations will be available in a "
                          "future update.")
            return
```
However, I still am of the opinion that in a stream this should set the correlation matrix to `None`

Reply:
added the test for stream